pandas/io/gbq.py

```diff
@@ -159,7 +159,47 @@ def get_credentials(self):
         if self.private_key:
             return self.get_service_account_credentials()
         else:
-            return self.get_user_account_credentials()
+            # Try to retrieve Application Default Credentials
+            project_id = self.project_id
+            credentials = self.get_application_default_credentials(project_id)
+            if not credentials:
+                credentials = self.get_user_account_credentials()
+            return credentials
+
+    @staticmethod
+    def get_application_default_credentials(project_id):  # pragma: no cover
```

Comment:
you need to follow the existing pattern, why is this a staticmethod? that's non-pythonic.

nested try/except is a no-no

just call your routine, if the default oauth imports raise you don't care, the user will get an import error.

only the addl imports (for getting default credientials) need to be caught (and return None).
you can